front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleTrainsListV2.tsx

```diff
@@ -27,28 +33,79 @@ function LoadingIfSearching({ isLoading, t }: { isLoading: boolean; t: TFunction
 type ImportTrainScheduleTrainsListProps = {
   trainsList: TrainSchedule[];
   rollingStocks: LightRollingStockWithLiveries[];
-  infraId: number;
   isLoading: boolean;
   timetableId: number;
 };
 
 const ImportTrainScheduleTrainsListV2 = ({
   trainsList,
   rollingStocks,
-  infraId,
   isLoading,
   timetableId,
 }: ImportTrainScheduleTrainsListProps) => {
   const { t } = useTranslation(['operationalStudies/importTrainSchedule']);
-  const { openModal } = useModal();
 
   const rollingStockDict = useMemo(
     () => keyBy(rollingStocks, (rollingStock) => rollingStock.name),
     [rollingStocks]
   );
 
+  const { refetch: refetchTimetable } =
+    osrdEditoastApi.endpoints.getTimetableById.useQuerySubscription({ id: timetableId });
+  // const [postPathFinding] = osrdEditoastApi.endpoints.postPathfinding.useMutation();
+  const [postTrainSchedule] = osrdEditoastApi.endpoints.postV2TrainSchedule.useMutation();
+
+  const [status, setStatus] = useState(initialStatus);
+  const dispatch = useAppDispatch();
+
+  async function launchTrainSchedules(payload: TrainScheduleForm) {
+    try {
+      await postTrainSchedule({ body: [payload] }).unwrap();
+    } catch (error) {
+      return false;
+    }
+
+    return true;
+  }
+  async function generateV2TrainSchedules() {
+    const payloads = generateV2TrainSchedulesPayload(trainsList, timetableId);
+
+    const trainsCount = payloads.length;
+    let successfulTrainsCount = 0;
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const payload of payloads) {
+      const success = await launchTrainSchedules(payload);
+      if (success) {
+        successfulTrainsCount += 1;
+      }
+    }
+    if (successfulTrainsCount > 0) {
+      dispatch(
+        setSuccess({
+          title: t('success'),
+          text: t('status.calculatingTrainScheduleCompleteAll_other', {
+            successfulTrainsCount,
+            trainsCount,
+          }),
```

Comment:
i18n can pluralize your string if you give it a count argument and remve the '_other' string in the key:
```suggestion
          text: t('status.calculatingTrainScheduleCompleteAll', {
            successfulTrainsCount: trainsCount - errorsNb,
            trainsCount,
            count: trainsNb - errorsNb,
          }),
```